syntax = "proto3";

package speech.v1;

// 音频流服务
service StreamService {
  rpc ProcessAudioStream(stream AudioChunk) returns (stream AudioResponse);
}

message AudioChunk {
  bytes audio_data = 1;
  int64 timestamp = 2;
  string session_id = 3;
}

message AudioResponse {
  bool success = 1;
  string message = 2;
}

// ASR服务
service ASRService {
  rpc Transcribe(AudioData) returns (TranscriptionResult);
}

message AudioData {
  bytes pcm_data = 1;
  int32 sample_rate = 2;
}

message TranscriptionResult {
  string text = 1;
  float confidence = 2;
} 